src/control/server/ctl_system.go

```diff
@@ -33,10 +33,58 @@ import (
 	"github.com/daos-stack/daos/src/control/system"
 )
 
-const (
-	memberStopTimeout   = 10 * retryDelay
-	prepShutdownTimeout = 10 * retryDelay
-)
+// updateMemberStatus requests registered harness to ping their instances (system
+// members) in order to determine IO Server process responsiveness. Update membership
+// appropriately.
+//
+// Each host address represents a gRPC server associated with a harness managing
+// one or more data-plane instances (DAOS system members).
+//
+// TODO: specify the ranks managed by the harness that should be started.
+func (svc *ControlService) updateMemberStatus(ctx context.Context, leader *IOServerInstance) error {
+	// exclude members with states that can't be updated with response check
+	statesToExclude := []system.MemberState{
+		system.MemberStateEvicted, system.MemberStateErrored,
+		system.MemberStateUnknown, system.MemberStateStopped,
+		system.MemberStateUnresponsive,
+	}
+	hostAddrs := svc.membership.Hosts(statesToExclude...)
+
+	svc.log.Debugf("updating response status for ranks on hosts: %v", hostAddrs)
+
+	// currently just update newly unresponsive or errored members
+	badRanks := make(map[uint32]system.MemberState)
+	for _, addr := range hostAddrs {
+		hResults, err := harnessAction(ctx, leader.msClient,
+			NewRemoteHarnessReq(HarnessQuery, addr))
```

Comment:
Finding myself kind of disliking this paradigm for making a request to harness members as I see more of it. I think it's a fine implementation under the covers, but it might be better to have a wrapper interface for callers with individual methods for each of these actions.

Reply:
will add a harnessClient interface